### ISSUE TITLE
TestMachinesLifecycle.testBasic fails for cloning

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#15441]
Page: Machines

This is our current top flake, and was breaking most PRs last week:
```
-> wait: ph_is_present(".pf-c-modal-box__title-text")
<- {'type': 'object', 'subtype': 'error', 'className': 'PhWaitCondTimeout', 'description': 'Error: condition did not become true\n    at <anonymous>:245:24', 'objectId': '6058255692305731313.5.1'}
Traceback (most recent call last):
  File "/var/home/martin/upstream/cockpit/test/verify/check-machines-lifecycle", line 141, in _testBasic
    b.wait_text(".pf-c-modal-box__title-text", "Create a clone VM based on subVmTest1")

testlib.Error: timeout
```
The [screenshot](https://logs.cockpit-project.org/logs/pull-15430-20210225-181217-907b9fef-rhel-8-4-distropkg/TestMachinesLifecycle-testBasic-rhel-8-4-distropkg-127.0.0.2-2301-FAIL.png) shows that the action menu gets opened and shows the "clone" action, but it never closes and the expected clone dialog never appears.

This reproduces locally just fine (I am using centos-8-stream). Let's use this issue to collect some debugging information.

**Comment by martinpitt** [comment]
One interesting point is this in the journal:
```
libvirt-dbus[2088]: g_variant_new_object_path: assertion 'g_variant_is_object_path (object_path)' failed
libvirt-dbus[2088]: g_variant_get_type: assertion 'value != NULL' failed
libvirt-dbus[2088]: g_variant_type_is_subtype_of: assertion 'g_variant_type_check (type)' failed
libvirt-dbus[2088]: g_variant_builder_add_value: assertion '!GVSB(builder)->expected_type || g_variant_is_of_type (value, GVSB(builder)->expected_type)' failed
libvirt-dbus[2088]: g_variant_new_object_path: assertion 'g_variant_is_object_path (object_path)' failed
libvirt-dbus[2088]: g_variant_get_type: assertion 'value != NULL' failed
libvirt-dbus[2088]: g_variant_type_is_subtype_of: assertion 'g_variant_type_check (type)' failed
libvirt-dbus[2088]: g_variant_builder_add_value: assertion '!GVSB(builder)->expected_type || g_variant_is_of_type (value, GVSB(builder)->expected_type)' failed
```
However, I think it's a red herring -- the process does not actually crash due to this, it's a glib critical. (1) it does not happen all the time, and (2) I get this in an interactive browser when I click on the clone action, and there I do get the dialog just fine.

**Comment by martinpitt** [comment]
* This happens with `TEST_SHOW_BROWSER=1` as well. While that is hanging waiting for the dialog, I can manually click on the "clone" menu action, and the test resumes.

 * This happens also with a much reduced test case, which makes iterating a lot faster, and reproducing this even more reliable (it now fails 100% for me):
```py
    def _testBasic(self, user=None, superuser=True, expect_empty_list=False):
        b = self.browser
        m = self.machine

        self.createVm("subVmTest1", running=False)
        self.login_and_go("/machines", user=user, superuser=superuser)
        self.performAction("subVmTest1", "clone")

        b.wait_text(".pf-c-modal-box__title-text", "Create a clone VM based on subVmTest1")
        b.click("footer button.pf-m-primary")
        b.wait_not_present(".pf-c-modal-box")
        self.waitVmRow("subVmTest1-clone")
```

 * This happens both on fresh VMs as well as re-using an already running one. So it is *not* about insufficient cleanup, and using a running one is fine for examining this (and much faster to iterate).
 * Adding sleep(1) before the performAction() is not sufficient. sleep(5) works. The main thing that visually happens in that time is that the "Create VM" and "Import VM" buttons appear, and the message below is printed. In other words, this is the libvirt initialization/detection time.

> warning: LIST_DOMAIN_SNAPSHOTS action failed for domain %s: %s /org/libvirt/QEMU/domain/_3f0f90e1_b6ab_4423_914b_fc064e702c8b {"problem":null,"name":"org.freedesktop.DBus.Error.UnknownMethod","message":"No such method 'ListDomainSnapshots'"}

 * Both the original and the reduced test reliably work if I add a `b.wait_visible("#create-new-vm:not(:disabled)")` right before the performAction(). This is of course not a fix, just diagnostics -- the kebab menu itself (and the Run button) should not appear until libvirt has initialized, same as the "Create/Import VM" buttons.

**Comment by martinpitt** [comment]
Fixed in PR #15442. I ran this 5 times in a row on my system, both the reduced and the full test.
